### PR TITLE
Move environment hook to pkg pt2

### PR DESCRIPTION
**Pull request by schuellerf**
This pull request includes:

- [✔️] adequate testing for the new functionality or fixed issue
- [✔️] adequate documentation informing people about the change such as
  - extended `make help`

functionally this PR does not change anything. It just
* moves the `environment_hook` to `pkg/splunk_logger` so it can be used by image-builder too
* unify behavior of github action and local test (by calling `make` from github action)
* **enables** testing of `pkg/splunk_logger` (this was never tested I guess)
* introduces `make coverage-report` as this is generated anyway and can be reviewed by humans like this

followup of #4301

### CODE REVIEW

**Comment by schuellerf** [comment]
just out of curiosity - is the `coverage.txt` (of the github actions) used somewhere?

File: Makefile [hunk 1]
```diff
@@ -234,8 +235,18 @@ worker-key-pair: ca
...
 unit-tests:
-	go test -race ./...
+	go test -race -covermode=atomic -coverprofile=coverage.txt -coverpkg=$$(go list ./... | grep -v rpmmd/test$ | tr "\n" ",") ./...
+	# sub module is not tested automatically
+	pushd pkg/splunk_logger
+	go test -race -covermode=atomic -coverprofile=../../coverage_splunk_logger.txt -coverpkg=$$(go list ./... | grep -v rpmmd/test$ | tr "\n" ",") ./...
+	popd
+
+.PHONY: coverage-report
+coverage-report: unit-tests
```
Comment (mvo5) [inline Makefile:247]: This is nice - I think it's would slightly nicer to split it into it's own commit though.

File: Makefile [hunk 2]
```diff
@@ -234,8 +235,18 @@ worker-key-pair: ca
 	rm /etc/osbuild-composer/worker-csr.pem
 
 .PHONY: unit-tests
+.ONESHELL:
 unit-tests:
-	go test -race ./...
+	go test -race -covermode=atomic -coverprofile=coverage.txt -coverpkg=$$(go list ./... | grep -v rpmmd/test$ | tr "\n" ",") ./...
+	# sub module is not tested automatically
```
Comment (mvo5) [inline Makefile:241]: When I read this comment I was confused for a second (okay, not just a second, a few more) and thought this was about "git" submodules. Maybe `# go modules with go.mod in subdirs are not tested automatically` or something like this?

But this makes me wonder if instead of adding special cases here we could just move `splunk_logger` into it's own git repo (e.g. `github.com/osbuild/splunk_logger` ) and then use the normal vendor mechanism to get it (just like any other dep). Then the tests for spluk_logger would run in the splunk_logger git repo (just like any other dep). Wdyt? Maybe too big of a change for such a tiny package but it would feel slightly cleaner to me (but maybe I'm missing something here) (adding @achilleas-k as it seems he and Diaa did some work on this)

Comment (schuellerf) [reply Makefile:241]: for sure I'll take your suggestion as new comment, thanks!

Totally agree, this should be separate. Also @croissanne emphasized several times, this should be split
…or even some library containing more than `splunk_logger`?
but for now the tests should at least run - if I mess with them ;)

Comment (achilleas-k) [reply Makefile]: 1. Call it a subpackage (Go calls them packages) to avoid confusion.
2. Let's split it out in a follow-up.

**Comment by achilleas-k** [comment]
> just out of curiosity - is the `coverage.txt` (of the github actions) used somewhere?

We have a coverity action running on a schedule but I don't know if it uses that file or does it's own thing.

File: Makefile (hunk 2 above)
Comment (mvo5) [reply Makefile]: If desired I can do the split tomorrow

Comment (croissanne) [reply Makefile]: Merging this and splitting sounds good to me. Took the liberty of creating https://github.com/osbuild/splunk_logger  :)

**Comment by schuellerf** [comment]
Thanks @croissanne for creating the newly split repo
I'll take care of
https://github.com/osbuild/splunk_logger/issues/1
which is a little more than just fixing a line :-/

can we merge this first (as this is waiting a while now)?
I'll immediately fix the new `splunk_logger` package afterwards (also here and in image-builder)

File: Makefile [hunk 3]
```diff
@@ -234,8 +235,17 @@ worker-key-pair: ca
 	rm /etc/osbuild-composer/worker-csr.pem
 
 .PHONY: unit-tests
+.ONESHELL:
 unit-tests:
-	go test -race ./...
+	go test -race -covermode=atomic -coverprofile=coverage.txt -coverpkg=$$(go list ./... | grep -v rpmmd/test$ | tr "\n" ",") ./...
+	# go modules with go.mod in subdirs are not tested automatically
+	cd pkg/splunk_logger
+	go test -race -covermode=atomic -coverprofile=../../coverage_splunk_logger.txt -coverpkg=$$(go list ./... | grep -v rpmmd/test$ | tr "\n" ",") ./...
```
Comment (achilleas-k) [inline Makefile:243]: The filtering out of rpmmd/test (`go list ./... | grep -v rpmmd/test$`) is a remnant from a long long time ago.  There is no package matching `rpmmd/test` anymore:
```
› go list ./... | grep -F rpmmd/test -c
0
```

Good opportunity for a clean up, I think.

**Comment by mvo5** [comment]
> Thanks @croissanne for creating the newly split repo I'll take care of [osbuild/splunk_logger#1](https://github.com/osbuild/splunk_logger/issues/1) which is a little more than just fixing a line :-/
> 
> can we merge this first (as this is waiting a while now)? I'll immediately fix the new `splunk_logger` package afterwards (also here and in image-builder)

Fwiw, I'm fine with merging this first but also curious - what is the extra work (beyond fixing the go.mod)? The workflow or is there more to it?

File: Makefile (hunk 3 above)
Comment (schuellerf) [reply Makefile:243]: the whole line will be removed by https://github.com/osbuild/splunk_logger/issues/1 soonish
but I can fix it for the line 240 here ;)

**Comment by schuellerf** [comment]
> Fwiw, I'm fine with merging this first but also curious - what is the extra work (beyond fixing the go.mod)? The workflow or is there more to it?

the things to be done are being collected in https://github.com/osbuild/splunk_logger/issues/1 right now

File: Makefile [hunk 4]
```diff
@@ -90,6 +90,7 @@ help:
 	@echo "                        (do this before pushing!)"
 	@echo "    lint:               Runs linters as close as github workflow as possible"
 	@echo "    process-templates:  Execute the OpenShift CLI to check the templates"
+	@echo "    coverage-report:    Run unit tests and generate HTML coverage reports"
```
[the remark this answers is not in the document]
Comment (schuellerf) [reply Makefile:93]: yes, which is not readable if you want to check locally on your computer (without codecov.io)

[the remark this answers is not in the document]
Comment (schuellerf) [reply Makefile:93]: so `make unit-tests` create the `txt` file and `make coverage-report` human readable `html` (for developers locally)

File: go.sum [hunk 5]
```diff
@@ -510,8 +510,8 @@ github.com/oracle/oci-go-sdk/v54 v54.0.0 h1:CDLjeSejv2aDpElAJrhKpi6zvT/zhZCZuXch
 github.com/oracle/oci-go-sdk/v54 v54.0.0/go.mod h1:+t+yvcFGVp+3ZnztnyxqXfQDsMlq8U25faBLa+mqCMc=
 github.com/osbuild/images v0.77.0 h1:O0Nv07M7b3YuY+c83/NFjWFachO+aH714zu7r+QkdsU=
 github.com/osbuild/images v0.77.0/go.mod h1:14LZWLSsQ02C/vZ+EzBkp+OcdjebnWDJ8moz8o/a0J4=
-github.com/osbuild/osbuild-composer/pkg/splunk_logger v0.0.0-20231117174845-e969a9dc3cd1 h1:UFEJIcPa46W8gtWgOYzriRKYyy1t6SWL0BI7fPTuVvc=
-github.com/osbuild/osbuild-composer/pkg/splunk_logger v0.0.0-20231117174845-e969a9dc3cd1/go.mod h1:z+WA+dX6qMwc7fqY5jCzESDIlg4WR2sBQezxsoXv9Ik=
+github.com/osbuild/osbuild-composer/pkg/splunk_logger v0.0.0-20240814102216-0239db53236d h1:r9BFPDv0uuA9k1947Jybcxs36c/pTywWS1gjeizvtcQ=
```
[the remark this answers is not in the document]
Comment (schuellerf) [reply go.sum:513]: same :-D

**Review by thozza — COMMENTED** [review]
Changes look fine to me. What's the end result WRT waiting for the external pkg to be used?

**Comment by schuellerf** [comment]
Obsolete due to #4316

**Comment by thozza** [comment]
> > Changes look fine to me. What's the end result WRT waiting for the external pkg to be used?
> 
> Not sure if I got your question right, but the `splunk_logger` is also used by `image-builder` or did you mean factoring out `splunk_logger` that's just more clean but can be done before or after this PR

I meant using the package from its own git repository and not from within this one.
